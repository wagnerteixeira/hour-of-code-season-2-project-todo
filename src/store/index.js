import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import layoutReducer from '../layout/redux/layoutReducer';
import todoReducer from '../todo/redux/todoReducer';

const reducers = combineReducers({  
  layout: layoutReducer,
  todo: todoReducer
});

export default createStore(reducers, applyMiddleware(thunk));
