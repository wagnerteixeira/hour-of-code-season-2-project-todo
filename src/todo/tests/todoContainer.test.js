import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import store from '../../store';
import _ from 'lodash';

import TodoContainer from '../todoContainer';
import { todoMock } from './todoMock';

configure({ adapter: new Adapter() });

describe('Test <TodoContainer />', function () {

    let component;
    const todoGroup = _.groupBy(todoMock.data, 'status');

    const todoList = todoGroup['TODO'] || [];
    const doingList = todoGroup['DOING'] || [];
    const doneList = todoGroup['DONE'] || [];

    beforeEach(() => {
        component = shallow(
            <Provider store={store}>
                <TodoContainer
                    todoList={todoList}
                    doingList={doingList}
                    doneList={doneList}
                />
            </Provider>
        );
    });

    it('render <TodoContainer /> ', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

});