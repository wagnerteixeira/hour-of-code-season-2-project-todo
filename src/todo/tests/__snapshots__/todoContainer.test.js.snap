// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TodoContainer /> render <TodoContainer />  1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Connect(WithStyles(PureComponent))
    doingList={
      Array [
        Object {
          "assignee": "Assignee 2",
          "description": "Description 2",
          "id": "2",
          "status": "DOING",
          "title": "Title 2",
        },
        Object {
          "assignee": "Assignee 5",
          "description": "Description 5",
          "id": "5",
          "status": "DOING",
          "title": "Title 5",
        },
      ]
    }
    doneList={
      Array [
        Object {
          "assignee": "Assignee 3",
          "description": "Description 3",
          "id": "3",
          "status": "DONE",
          "title": "Title 3",
        },
        Object {
          "assignee": "Assignee 6",
          "description": "Description 6",
          "id": "6",
          "status": "DONE",
          "title": "Title 6",
        },
      ]
    }
    todoList={
      Array [
        Object {
          "assignee": "Assignee 1",
          "description": "Description 1",
          "id": "1",
          "status": "TODO",
          "title": "Title 1",
        },
        Object {
          "assignee": "Assignee 4",
          "description": "Description 4",
          "id": "4",
          "status": "TODO",
          "title": "Title 4",
        },
        Object {
          "assignee": "Assignee 7",
          "description": "Description 7",
          "id": "7",
          "status": "TODO",
          "title": "Title 7",
        },
      ]
    }
  />
</ContextProvider>
`;
